docs/process/process_areas/implementation/guidance/cpp/coding_guidelines.rst

```diff
@@ -0,0 +1,96 @@
+..
+   # *******************************************************************************
+   # Copyright (c) 2025 Contributors to the Eclipse Foundation
+   #
+   # See the NOTICE file(s) distributed with this work for additional
+   # information regarding copyright ownership.
+   #
+   # This program and the accompanying materials are made available under the
+   # terms of the Apache License Version 2.0 which is available at
+   # https://www.apache.org/licenses/LICENSE-2.0
+   #
+   # SPDX-License-Identifier: Apache-2.0
+   # *******************************************************************************
+
+Writing C++ Code
+################
+
+.. gd_guidl:: Coding Guidelines CPP
+   :id: gd_guidl__cpp_coding_guidelines
+   :status: valid
+   :complies: std_req__iso26262__software_1, std_req__iso26262__software_2, std_req__iso26262__software_3
+
+Coding Guidelines
+*****************
+Most programming languages are not specifically designed to be used within safety critical systems. If they are still applied in this context, *coding guidelines* are required. In those guidelines e.g. a safe subset of the language is defined which can be used safely and is not known to be error-prone. Another approach could be that those guidelines require a SW which includes only low complexity code. This means by using those guidelines the risk for systematic error, when using this language, can be minimized.
+
+MISRA C++:2023
+--------------
+
+These guidelines were released on November 2023, and they replace the *AUTOSAR guidelines for C++14* and the *MISRA C++:2008 guidelines*.
+
+*MISRA C++:2023* is designed to be used with *C++17*. If these were used in older C++ versions like *C++14* or *C++11*, it would mean that additional or different guidelines would need to be applied. For example, from *MISRA C++:2008* and/or *AUTOSAR* some rules have been dropped because they address issues which are no longer relevant for *C++17*. The same would apply if a newer C++ standard would be used like for example *C++20* or *C++23*.
+
+Code Formatter
+==============
+Appropriate code formatting provides a source code which is easier to read and better to understand as it follows a well defined, known standard. That is why usually code formatters are applied. For this project *Clang-Format* shall be used.
+
+Code Style
+==========
+A *code style* describes which patterns, elements, conventions or best practices should be used during implementation.
+
+Examples are:
+
+* naming convention
+* function calls
+* initialization of variables.
+
+For this project an adapted code style based on the `google style guide <https://google.github.io/styleguide/cppguide.html>`_ shall be used.
+
+Language Feature Selection
+==========================
+
+Basically, all C++17 Features which are allowed by the *MISRA C++:2023* guidelines can be used.
```

Comment:
Mention deviation process? e.g. "any and all deviations need to be confirmed by language experts and process team. Detailed workflow will follow on demand".